behaviors/magic-button.js

```diff
@@ -30,10 +43,46 @@ const _ = require('lodash'),
       }
 
       return path;
+    },
+
+    /**
+     * transform rich text into a hyphen-delineated slug
+     * @param {string} data
+     * @returns {string}
+     */
+    toSlug: function (data) {
+      // remove EVERYTHING from the slug, then run it through speakingurl
+      return speakingurl(toPlainText(stripUnicode(data)), {
+        custom: {
+          _: '-' // convert underscores to hyphens
+        }
+      });
     }
   };
 
 /**
+ * Removes all unicode from string
+ * @param {string} str
+ * @returns {string}
+ */
+function stripUnicode(str) {
```

Comment:
Do we have a more Util-y place for this guy to live? Could see this being handy elsewhere.